src/main/java/io/cdap/plugin/successfactors/source/transport/SuccessFactorsTransporter.java

```diff
@@ -153,8 +153,23 @@ public Response retrySapTransportCall(URL endpoint, String mediaType) throws IOE
    */
   private Response transport(URL endpoint, String mediaType) throws IOException, TransportException {
     OkHttpClient enhancedOkHttpClient = getConfiguredClient().build();
-    Request req = buildRequest(endpoint, mediaType);
-
+    Request req = null;
+    if (config.getAuthType().equals("basicAuth")) {
+      req = buildRequest(endpoint, mediaType);
+    } else if (config.getAuthType().equals("oAuth2")) {
```

Comment:
make use of `else` instead of `else if` OR `proxy` will also be part of it.

Reply:
done